### PR TITLE
perf(auditlog): add event to grouped datetime index

### PR DESCRIPTION
**Pull request by JoshFerge**
we added an index to auditlogentry in #27508 to speed up the general queries, but the index doesn't help when sorting by event, which one can do in the UI. 

This PR adds event into the combined index which should speed up those queries while keeping the general org query fast too.

### CODE REVIEW

File: src/sentry/migrations/0228_update_auditlog_index_with_entry.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    dependencies = [
+        ("sentry", "0227_backfill_visits"),
+    ]
+
+    operations = [
+        migrations.SeparateDatabaseAndState(
+            database_operations=[
+                migrations.RunSQL(
+                    """DROP INDEX CONCURRENTLY IF EXISTS sentry_audi_organiz_c8bd18_idx;""",
```
Comment (JoshFerge) [inline src/sentry/migrations/0228_update_auditlog_index_with_entry.py:35]: this was the index added in #27508

**Review by markstory — APPROVED** [review]
Looks good.

**Review by wedamija — COMMENTED** [review]
I just want to make sure I understand the query patterns here. Are you doing queries like

```
select *
from sentry_auditlogentry
where organization_id = <x>
order by event, datetime
```

```
select *
from sentry_auditlogentry
where organization_id = <x>
order by event
```

```
select *
from sentry_auditlogentry
where organization_id = <x>
order by datetime
```

This new index will work for the first two, but not the third

**Comment by JoshFerge** [comment]
@wedamija i see, yup. misunderstood our slack thread. we'll need both indexes to keep both query patterns fast. thanks for the catch.

**Review by wedamija — APPROVED** [review]

**Pull request merged**